### PR TITLE
BTAT-3001 implemented check direct debit exists calling DES

### PR DESCRIPTION
**Pull request by HarryGull**

### CODE REVIEW

File: app/connectors/FinancialDataConnector.scala [hunk 1]
```diff
@@ -34,6 +34,9 @@ class FinancialDataConnector @Inject()(val http: HttpClient, val appConfig: Micr
   private[connectors] def financialDataUrl(regime: TaxRegime) =
     s"${appConfig.desUrl}/enterprise/financial-data/${regime.idType}/${regime.id}/${regime.regimeType}"
 
+  private[connectors] def directDebitUrl(vrn: String) =
```
Comment (lewishay) [inline app/connectors/FinancialDataConnector.scala:37]: There is no unit test for this function.

File: app/connectors/FinancialDataConnector.scala [hunk 2]
```diff
@@ -49,4 +52,20 @@ class FinancialDataConnector @Inject()(val http: HttpClient, val appConfig: Micr
         error
     }
   }
+
+  def checkDirectDebitExists(vrn: String)
```
Comment (lewishay) [inline app/connectors/FinancialDataConnector.scala:56]: There are no unit tests for this function.

File: test/mocks/services/MockFinancialTransactionsService.scala [hunk 3]
```diff
@@ -17,7 +17,7 @@
 package mocks.services
 
 import connectors.httpParsers.FinancialTransactionsHttpParser.HttpGetResult
-import models.{FinancialDataQueryParameters, FinancialTransactions, TaxRegime}
+import models.{DirectDebits, FinancialDataQueryParameters, FinancialTransactions, TaxRegime}
```
Comment (lewishay) [inline test/mocks/services/MockFinancialTransactionsService.scala]: Unused import.

File: test/services/FinancialTransactionsServiceSpec.scala [hunk 4]
```diff
@@ -180,4 +180,53 @@ class FinancialTransactionsServiceSpec extends SpecBase with MockFinancialDataCo
...
+      setupMockCheckDirectDebitExists(vrn)(successResponse)
+
+      setupMockAuditEventResponse(DirectDebitsCheckResponseAuditModel(vrn, hasDirectDebit = true))
+      setupMockAuditEventResponse(DirectDebitCheckRequestAuditModel(vrn))
+
+      val actual: Either[ErrorResponse, Boolean] = await(TestFinancialTransactionService.checkDirectDebitExists(vrn))
+
+      actual shouldBe successResponse
+
+      verifyAuditEvent(DirectDebitCheckRequestAuditModel(vrn))
```
Comment (lewishay) [inline test/services/FinancialTransactionsServiceSpec.scala:200]: These _verifyAuditEvent_ functions don't return anything and are after the test assertion. Are they necessary?

**Review by lewishay — CHANGES_REQUESTED** [review]
Could you also please update the scoverage settings in build.sbt to increase the coverage minimum to 95 and set coverageFailOnMinimum to true.

File: test/services/FinancialTransactionsServiceSpec.scala (hunk 4 above)
Comment (HarryGull) [reply test/services/FinancialTransactionsServiceSpec.scala:200]: They verify the method was called during the test

File: build.sbt [hunk 5]
```diff
@@ -46,9 +46,10 @@ lazy val coverageSettings: Seq[Setting[_]] = {
 
   Seq(
     ScoverageKeys.coverageExcludedPackages := excludedPackages.mkString(";"),
-    ScoverageKeys.coverageMinimum := 90,
+    ScoverageKeys.coverageMinimum := 95,
     ScoverageKeys.coverageFailOnMinimum := false,
-    ScoverageKeys.coverageHighlighting := true
+    ScoverageKeys.coverageHighlighting := true,
+    ScoverageKeys.coverageFailOnMinimum := true
```
Comment (lewishay) [inline build.sbt]: This property is being set twice. Could you remove the "false" line please.

**Review by lewishay — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 5 above)
Comment (HarryGull) [reply build.sbt]: done

**Pull request merged**